### ISSUE TITLE
Speed up CMake configuration checks

### ISSUE THREAD

**Issue by jbcoe** [jbcoe/indirect_value#11]
https://github.com/cristianadam/cmake-checks-cache suggests that we can speed up the CMake build.

```
time ./scripts/build.py -t --clean
```

on my (rather old) laptop takes

```
real	1m26.180s
user	0m9.613s
sys	0m1.904s
```

**Comment by jbcoe** [comment]
Work so far looks promising:

```
mkdir external
git submodule add git@github.com:cristianadam/cmake-checks-cache.git external/cmake-checks-cache
cmake . -G Ninja -B _build -DCMAKE_MODULE_PATH=`pwd`/external/cmake-checks-cache
```

makes subsequent invocations of

```
cmake . -G Ninja -B _build -DCMAKE_MODULE_PATH=`pwd`/external/cmake-checks-cache
```

very fast.

`scripts/build.py` will need updating and we need to decide how to handle a pre-cmake submodule

**Comment by Twon** [comment]
I've given this module a go without much luck.  On investigation, it appears that even when I pass the module in on the command line (via -DCMAKE_MODULE_PATH=<path to module>) the checks are not being called.  Initially, I added message statements to the macro definitions, but when this proved fruitless I redirected the trace output to a file as such:

```bash
 cmake ../ -DCMAKE_MODULE_PATH=<path to module>-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON --trace >> build.log 2>&1
```

Looking at the logs I see no calls to any of the expected check calls.  At this point, I am not sure why? It may be the project is not doing anything complicated enough to invoke the checks.  But further research is needed to confirm this.
